packages/calcite-components/src/components/block/block.scss

```diff
@@ -23,6 +23,10 @@
   @apply justify-start;
 }
 
+.header-padding {
```

Comment:
nitpick: maybe this should be named something like `.header--present` or `.header--enabled`?